glib/src/gstring.rs

```diff
@@ -14,6 +14,326 @@ use std::ptr;
 use std::slice;
 use std::string::String;
 
+// rustdoc-stripper-ignore-next
+/// Representaion of a borrowed [`GString`].
+///
+/// This type is very similar to [`std::ffi::CStr`], but with one added constraint: the string
+/// must also be valid UTF-8.
+#[derive(Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+#[repr(transparent)]
+pub struct GStr([u8]);
```

Comment:
Ah nice, I have a commit that adds such a thing locally too :) Why not

```suggestion
pub struct GStr(str);
```

though?

Reply:
Sure